src/tox_lsr/test_scripts/runcodeql.sh

```
@@ -0,0 +1,115 @@
+
+# Load language configuration
+codeql resolve queries python-code-scanning.qls --format=bylanguage
+
+codeql resolve queries python-security-and-quality.qls --format=bylanguage
+
+codeql resolve languages --format=betterjson --extractor-options-verbosity=4
+
+# Setup Python dependencies
+# $CODEQLACTIONDIR/python-setup/install_tools.sh
+# Remove "--user" from "pip install" to workaround this error.
+# ERROR: Can not perform a '--user' install. User site-packages are
+# not visible in this virtualenv.
+sed -e "s/pip install --user/pip install/" \
+$CODEQLACTIONDIR/python-setup/install_tools.sh > $WORKDIR/install_tools.sh
+bash -x $WORKDIR/install_tools.sh
+
+codeql database init --db-cluster $DBDIR --source-root=$TOPDIR \
+    --language=python
+
+# Setup environment variables
+export CODEQL_WORKFLOW_STARTED_AT=$( date -Iseconds )
+export CODEQL_RAM=5919
+export CODEQL_THREADS=2
+
+# Extracting python
+codeql database trace-command $DBDIR/python -- \
+    $( dirname "$CODEQLPATH" )/python/tools/autobuild.sh
+
+# Finalizing python
+codeql database finalize --finalize-dataset --threads=$CODEQL_THREADS \
+    --ram=$CODEQL_RAM $DBDIR/python
+
+# Running queries for python
+codeql database run-queries --ram=$CODEQL_RAM --threads=$CODEQL_THREADS \
+    $DBDIR/python --min-disk-free=1024 \
+    -v python-security-and-quality.qls
+
+# Interpreting results for python
+codeql database interpret-results --threads=$CODEQL_THREADS \
+    --format=sarif-latest -v --output=$RESULTS/python.sarif \
+    --no-sarif-add-snippets --print-diagnostics-summary \
+    --print-metrics-summary --sarif-group-rules-by-pack \
+    --sarif-add-query-help --sarif-category /language:python \
+    --sarif-add-baseline-file-info $DBDIR/python \
+    python-security-and-quality.qls
+
+codeql database print-baseline $DBDIR/python
+
+echo "CodeQL result file on $ROLE: $RESULTS/python.sarif"
+
+JQPATH=$( which jq 2> /dev/null )
+if [ $? -ne 0 ]; then
+    echo "WARNING: please install jq package"
```

move this check to near the top so the script fails early if jq not found